zerver/tests/test_messages.py

```diff
@@ -1443,12 +1443,22 @@ def test_message_to_announce(self) -> None:
 
         # Cross realm bots should be allowed.
         notification_bot = get_system_bot("notification-bot@zulip.com")
-        result = self.api_post(notification_bot.email,
-                               "/json/messages", {"type": "stream",
-                                                  "to": stream_name,
-                                                  "client": "test suite",
-                                                  "content": "Test message",
-                                                  "topic": "Test topic"})
+        result = self.api_post(notification_bot.email, "/api/v1/messages", {"type": "stream",
```

Comment:
Can you please explain why we're converting "/json/" to "/api/v1/"? I vaguely remember something about this coming up on the Zulip Community, a brief explanation would be great :slightly_smiling_face:. Also, maybe we should do this renaming in a separate prep commit, though I'm sure about if it's required. @timabbott any comments?

Reply:
@Hypro999 /api/v1/ endpoint is used for bot requests and not /json. The old test was buggy, hence this fix. I also thought to make it a separate commit. But this test fix was really small and related to my new tests and logic so I included it along this same commit.

I can separate the commits if required. @timabbott ?

Reply:
Ah, thanks for clarifying this Mohit!

> But this test fix was really small and related to my new tests and logic so I included it along this same commit.

Agreed, but separate commits might be the standard.

Reply:
I just pushed two separate commits.

Reply:
Yes, doing separate commit is much better, thanks for splitting this.  As a general rule, changes that has a different reason/purpose belong in their own commit, regardless of how small they are, because it makes both code review faster and understanding the history when debugging (Which I do all the time to understand why the code is the way it is) a lot easier.